src/uu/dd/src/dd.rs

```diff
@@ -0,0 +1,1296 @@
+
+        if let Some(ct) = i.cflags.ctable {
+            apply_ct(&mut buf, ct);
+        }
+
+        let buf = unblock(buf, cbs);
+
+        Ok(buf)
+    } else {
+        // The following error should not happen, as it results from
+        // insufficient command line data. This case should be caught
+        // by the parser before making it this far.
+        // Producing this error is an alternative to risking an unwrap call
+        // on 'cbs' if the required data is not provided.
+        Err(Box::new(InternalError::InvalidConvBlockUnblockCase))
+    }
+}
+
+fn read_helper<R: Read>(
+    i: &mut Input<R>,
+    bsize: usize,
+) -> Result<(ReadStat, Vec<u8>), Box<dyn Error>> {
+    // Local Predicate Fns -----------------------------------------------
+    #[inline]
+    fn is_conv<R: Read>(i: &Input<R>) -> bool {
+        i.cflags.ctable.is_some()
+    }
+    #[inline]
+    fn is_block<R: Read>(i: &Input<R>) -> bool {
+        i.cflags.block.is_some()
+    }
+    #[inline]
+    fn is_unblock<R: Read>(i: &Input<R>) -> bool {
+        i.cflags.unblock.is_some()
+    }
+    // Local Helper Fns -------------------------------------------------
+    #[inline]
+    fn perform_swab(buf: &mut [u8]) {
+        let mut tmp;
+
+        for base in (1..buf.len()).step_by(2) {
+            tmp = buf[base];
+            buf[base] = buf[base - 1];
+            buf[base - 1] = tmp;
+        }
+    }
+    // ------------------------------------------------------------------
+    // Read
+    let mut buf = vec![BUF_INIT_BYTE; bsize];
+    let mut rstat = match i.cflags.sync {
+        Some(ch) => i.fill_blocks(&mut buf, ch)?,
+        _ => i.fill_consecutive(&mut buf)?,
+    };
+    // Return early if no data
+    if rstat.reads_complete == 0 && rstat.reads_partial == 0 {
+        return Ok((rstat, buf));
+    }
+
+    // Perform any conv=x[,x...] options
+    if i.cflags.swab {
+        perform_swab(&mut buf);
+    }
+    if is_conv(i) || is_block(i) || is_unblock(i) {
+        let buf = conv_block_unblock_helper(buf, i, &mut rstat)?;
+        Ok((rstat, buf))
+    } else {
+        Ok((rstat, buf))
+    }
+}
+
+fn print_io_lines(update: &ProgUpdate) {
+    eprintln!(
+        "{}+{} records in",
+        update.reads_complete, update.reads_partial
+    );
+    if update.records_truncated > 0 {
+        eprintln!("{} truncated records", update.records_truncated);
+    }
+    eprintln!(
+        "{}+{} records out",
+        update.writes_complete, update.writes_partial
+    );
+}
+fn make_prog_line(update: &ProgUpdate) -> String {
+    let btotal_metric = Byte::from_bytes(update.bytes_total)
+        .get_appropriate_unit(false)
+        .format(0);
+    let btotal_bin = Byte::from_bytes(update.bytes_total)
+        .get_appropriate_unit(true)
+        .format(0);
+    let safe_millis = cmp::max(1, update.duration.as_millis());
+    let xfer_rate = Byte::from_bytes(1000 * (update.bytes_total / safe_millis))
```

Comment:
I'm guessing this means `transfer_rate`?

Reply:
Sure is. I made this change.